packages/legend-query/src/components/QueryBuilderFunctionPanel.tsx

```diff
@@ -0,0 +1,653 @@
+ * limitations under the License.
+ */
+
+import { observer } from 'mobx-react-lite';
+import {
+  type TooltipPlacement,
+  type TreeNodeContainerProps,
+  type TreeNodeViewProps,
+  ExpandIcon,
+  FolderIcon,
+  FolderOpenIcon,
+  InfoCircleIcon,
+  SubjectIcon,
+  Tooltip,
+  ViewHeadlineIcon,
+  ChevronDownIcon,
+  ChevronRightIcon,
+  PURE_FunctionIcon,
+  CompressIcon,
+  TreeView,
+  BlankPanelContent,
+  MenuContentItem,
+  DropdownMenu,
+  MenuContent,
+  CheckIcon,
+  MenuContentItemIcon,
+  MenuContentItemLabel,
+  MoreVerticalIcon,
+  clsx,
+} from '@finos/legend-art';
+import {
+  type QueryBuilderPackageElementTreeNodeData,
+  type QueryBuilderFunctionDragSource,
+  QUERY_BUILDER_FUNCTION_TREE_DND_TYPE,
+  generateQueryBuilderPackableElementTreeNodeData,
+  getTreeChildNodes,
+  generateFunctionSignature,
+} from '../stores/QueryFunctionsState';
+import { useDrag, useDragLayer } from 'react-dnd';
+import {
+  ConcreteFunctionDefinition,
+  Package,
+  ROOT_PACKAGE_NAME,
+} from '@finos/legend-graph';
+import { getEmptyImage } from 'react-dnd-html5-backend';
+import { useEffect, useState } from 'react';
+import type { QueryBuilderState } from '../stores/QueryBuilderState';
+import { getMultiplicityDescription } from './shared/QueryBuilderUtils';
+
+const QueryBuilderFunctionInfoTooltip: React.FC<{
+  element: ConcreteFunctionDefinition;
+  children: React.ReactElement;
+  placement?: TooltipPlacement | undefined;
+}> = (props) => {
+  const { element, children, placement } = props;
+  return (
+    <Tooltip
+      arrow={true}
+      {...(placement !== undefined ? { placement } : {})}
+      classes={{
+        tooltip: 'query-builder__tooltip',
+        arrow: 'query-builder__tooltip__arrow',
+        tooltipPlacementRight: 'query-builder__tooltip--right',
+      }}
+      TransitionProps={{
+        timeout: 0,
+      }}
+      title={
+        <div className="query-builder__tooltip__content">
+          <div className="query-builder__tooltip__item">
+            <div className="query-builder__tooltip__item__label">
+              Function Name
+            </div>
+            <div className="query-builder__tooltip__item__value">
+              {element.path}
+            </div>
+          </div>
+          <div className="query-builder__tooltip__item">
+            <div className="query-builder__tooltip__item__label">
+              Parameters
+            </div>
+            <div className="query-builder__tooltip__item__value">
+              {element.parameters
+                .map(
+                  (t) =>
+                    `${t.name}: ${
+                      t.type.value.name
+                    }${getMultiplicityDescription(t.multiplicity)}`,
+                )
+                .join('; ')}
+            </div>
+          </div>
+          <div className="query-builder__tooltip__item">
+            <div className="query-builder__tooltip__item__label">
```

Comment:
What do we show when there are no docs?

Reply:
<img width="277" alt="Screen Shot 2022-03-01 at 1 22 07 PM" src="https://user-images.githubusercontent.com/73408381/156226467-e760a2fe-1c53-417b-b7a7-a1388ef473a5.png">